### PR TITLE
Use a TwoPaneScaffold for ReceivedScreen

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #126

### CODE REVIEW

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+private fun DetailPane(navigator: ThreePaneScaffoldNavigator<String>) {
+    var lastSelectedTransfer by rememberSaveable { mutableStateOf<String?>(null) }
+
+    val transferUuid = navigator.currentDestination?.content ?: lastSelectedTransfer
+    navigator.currentDestination?.content?.let { lastSelectedTransfer = it }
+
+    if (transferUuid == null) {
+        NoSelectionEmptyState()
+        return
+    }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt:68]: Use if else instead

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+private fun ListPane(navigator: ThreePaneScaffoldNavigator<String>) {
+    ReceivedScreen(
+        navigateToDetails = { transferUuid -> navigator.navigateTo(ListDetailPaneScaffoldRole.Detail, transferUuid) },
+    )
+}
+
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+@Composable
+private fun DetailPane(navigator: ThreePaneScaffoldNavigator<String>) {
+    var lastSelectedTransfer by rememberSaveable { mutableStateOf<String?>(null) }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt:60]:
```suggestion
    var lastSelectedTransferUuid by rememberSaveable { mutableStateOf<String?>(null) }
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    val transferUuid = navigator.currentDestination?.content ?: lastSelectedTransfer
+    navigator.currentDestination?.content?.let { lastSelectedTransfer = it }
+
+    if (transferUuid == null) {
+        NoSelectionEmptyState()
+        return
+    }
+
+    val navigateBackCallback: () -> Unit = { navigator.navigateBack() }
+    val navigateBack: (() -> Unit)? = if (navigator.canNavigateBack()) navigateBackCallback else null
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt:71]: This piece of code is a dupplication, isn't it?
Wouldn't it be nice to make it a utils function?

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    )
+}
+
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+@Composable
+private fun DetailPane(navigator: ThreePaneScaffoldNavigator<String>) {
+    var lastSelectedTransfer by rememberSaveable { mutableStateOf<String?>(null) }
+
+    val transferUuid = navigator.currentDestination?.content ?: lastSelectedTransfer
+    navigator.currentDestination?.content?.let { lastSelectedTransfer = it }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt:63]: Why do we need to do this?

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt (hunk 4 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/received/ReceivedScreenWrapper.kt]: Looks like an oversight.

**Review by sirambd — APPROVED** [review]

**Pull request merged**